SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs

```diff
@@ -0,0 +1,258 @@
+            }
+            _noiseOctaves = (int)NoiseOctavesSlider.Value;
+            _noiseFrequency = new Vector2((float)NoiseFrequencySlider.Value, (float)NoiseFrequencySlider.Value);
+        }
+
+        private void ComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+            if (NoiseTypeComboBox.SelectedValue as string == "Fractal Sum")
+            {
+                _noiseType = TurbulenceEffectNoise.FractalSum;
+            }
+            else
+            {
+                _noiseType = TurbulenceEffectNoise.Turbulence;
+            }
+        }
+
+        #endregion
+
+        private bool _initialized = false;
+
+        private int _noiseOctaves = 4;
+        private Vector2 _noiseFrequency = new Vector2(0.05f, 0.05f);
+        private TurbulenceEffectNoise _noiseType = TurbulenceEffectNoise.Turbulence;
+
+        private TurbulenceEffect _noiseEffect = null;
+        private LuminanceToAlphaEffect _noiseLuminanceEffect = null;
+        private ManagedSurface _noiseSurface;
+
+        private ManagedSurface _contentSurface = null;
+        private CompositeEffect _combinedEffect = null;
+        private CompositionEffectBrush _combinedEffectBrush = null;
+        private SpriteVisual _outputVisual = null;
+
+        private Vector2 EffectSize
+        {
+            get
+            {
+                if (!double.IsNaN(Placeholder.ActualWidth) && !double.IsNaN(Placeholder.ActualHeight) && Placeholder.ActualWidth > 0 && Placeholder.ActualHeight > 0)
+                {
+                    return new Vector2((float)Placeholder.ActualWidth, (float)Placeholder.ActualHeight);
+                }
+                else
+                {
+                    return Vector2.Zero;
+                }
+            }
+        }
+
+        private async Task SetupEffect()
+        {
+            DisposeEffect();
+
+            _noiseEffect = new TurbulenceEffect();
+            _noiseEffect.Octaves = _noiseOctaves;
+            _noiseEffect.Frequency = _noiseFrequency;
+            _noiseEffect.Noise = _noiseType;
+            _noiseEffect.Tileable = true;
+            _noiseEffect.Size = EffectSize;
+
+            _noiseLuminanceEffect = new LuminanceToAlphaEffect();
+            _noiseLuminanceEffect.Source = _noiseEffect;
+
+            _noiseSurface = ImageLoader.Instance.LoadInteropSurface(EffectSize.ToSize(), NoiseDrawHandler);
+            if (_noiseEffect == null)
+            {
+                DisposeEffect();
+                return;
+            }
+            _noiseSurface.Brush.Stretch = CompositionStretch.Fill;
+
+            var placeholderVisual = ElementCompositionPreview.GetElementVisual(Placeholder);
+            var compositor = placeholderVisual.Compositor;
+
+            _contentSurface = await ImageLoader.Instance.LoadFromUriAsync(new Uri("ms-appx:///Assets/Landscapes/Landscape-1.jpg"));
```

Comment:
You should avoid re-creating the contentSurface and Effects here every time the slider updates.  Please separate this into one time initialization of stuff that isn't affected by UI updates and only update the noise surface when the variables are changed.